src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,206 @@
+        OperatorIndex memory operatorIndexToCheck = operatorIdToIndexHistory[operatorId][quorumNumber][index];
+
+        //blocknumber must be before the "index'th" entry's toBlockNumber
+        require(blockNumber <= operatorIndexToCheck.toBlockNumber, "IndexRegistry.getOperatorIndexForQuorumAtBlockNumberByIndex: provided index is too far in the future for provided block number");
+       
+        //if there is an index update before the "index'th" update, the blocknumber must be after than the previous entry's toBlockNumber
+        if(index != 0){
+            OperatorIndex memory previousOperatorIndex = operatorIdToIndexHistory[operatorId][quorumNumber][index - 1];
+            require(blockNumber > previousOperatorIndex.toBlockNumber, "IndexRegistry.getOperatorIndexForQuorumAtBlockNumberByIndex: provided index is too far in the past for provided block number");
+        }
+        return operatorIndexToCheck.index;
+    }
+
+    /**
+     * @notice Looks up the number of total operators for `quorumNumber` at the specified `blockNumber`.
+     * @param quorumNumber is the quorum number for which the total number of operators is desired
+     * @param blockNumber is the block number at which the total number of operators is desired
+     * @param index is the index of the entry in the dynamic array `totalOperatorsHistory[quorumNumber]` to read data from
+     */
+    function getTotalOperatorsForQuorumAtBlockNumberByIndex(uint8 quorumNumber, uint32 blockNumber, uint32 index) external view returns (uint32){
+        OperatorIndex memory operatorIndexToCheck = totalOperatorsHistory[quorumNumber][index];
+
+        //blocknumber must be before the "index'th" entry's toBlockNumber
+        require(blockNumber <= operatorIndexToCheck.toBlockNumber, "IndexRegistry.getTotalOperatorsForQuorumAtBlockNumberByIndex: provided index is too far in the future for provided block number");
+        
+        //if there is an index update before the "index'th" update, the blocknumber must be after than the previous entry's toBlockNumber
+        if (index != 0){
+            OperatorIndex memory previousOperatorIndex = totalOperatorsHistory[quorumNumber][index - 1];
+            require(blockNumber > previousOperatorIndex.toBlockNumber, "IndexRegistry.getTotalOperatorsForQuorumAtBlockNumberByIndex: provided index is too far in the past for provided block number");
+        }
+        return operatorIndexToCheck.index;
+    }
+
+    function totalOperatorsForQuorum(uint8 quorumNumber) external view returns (uint32){
+        return totalOperatorsHistory[quorumNumber][totalOperatorsHistory[quorumNumber].length - 1].index;
+    }
+
+
+    /// @notice Returns the current number of operators of this service.
+    function totalOperators() external view returns (uint32){
+        return uint32(globalOperatorList.length);
+    }
+
+
+    function _updateTotalOperatorHistory(uint8 quorumNumber, uint32 numOperators) internal {
```

Comment:
should probably have some documentation on this just explaining the intention

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/b03c92cfbb9bfd3366ad0bb1c11c20d3cedf29c1